### DISCUSSION TITLE
Calling ReadDbmsOutput logs error "driver: remove argument from query"

### DISCUSSION THREAD

**Discussion by sindy-senorita** [godror/godror#163] [Q&A] [answered]
We use `godror.ReadDbmsOutput(ctx, &buf, conn)` to read and process data from the dbms output.
It produces error log "driver: remove argument from query". 
From golang doc,[ that error is from NamedValueChecker](https://golang.org/pkg/database/sql/driver/#:~:text=ErrRemoveArgument%20may%20be%20returned%20from%20NamedValueChecker%20to%20instruct%20the%20sql%20package%20to%20not%20pass%20the%20argument%20to%20the%20driver%20query%20interface). Our internal logging library catches and logs the error. Caught in query `BEGIN DBMS_OUTPUT.get_lines(:1, :2); END;`
The call and result processing goes well without failure, but the error is trashing the log.

Since the logging library is out of our control, do you have any idea of how to remove the log?
or is there any other way to read dbms output without triggering the log?

Thanks!

**Comment by tgulacsi** [comment, marked as answer]
`statement.CheckNamedValue` returns driver.ErrRemoveArgument. 

The doc says:
```
// ErrRemoveArgument may be returned from NamedValueChecker to instruct the           
// sql package to not pass the argument to the driver query interface.                
// Return when accepting query specific options or structures that aren't             
// SQL query arguments.                                                
```

So, this is a totally legit return value.

What does catch this, and why?

**Reply by sindy-senorita** [reply]
We log db operations on driver level using https://github.com/simukti/sqldb-logger . In that repo, `connection.go` [logs the error](https://github.com/simukti/sqldb-logger/blob/c35f87e285f2b67e3c44529f89be172664f2e4c9/connection.go#L233). 

Do you know why NamedValueChecker decides to `instruct the sql package to not pass the argument`? 

PS: we are passing `sql.Open("godror", [the dsn URL])` directly to the logger, and the logger simply logs and passes everything through

**Reply by tgulacsi** [reply]
Yes, some parameters are treated as options (godror.PLSQLArray, godror.PrefetchCount, godror.FetchArraySize...)

https://github.com/simukti/sqldb-logger/blob/c35f87e285f2b67e3c44529f89be172664f2e4c9/connection.go#L233 is simply wrong, should not treat ErrRemoveArgument as an error.
